src/query/catalog/src/table_context.rs

```diff
@@ -84,8 +84,8 @@ pub trait TableContext: Send + Sync {
     fn get_partition(&self) -> Option<PartInfoPtr>;
     fn get_partitions(&self, num: usize) -> Vec<PartInfoPtr>;
     fn set_partitions(&self, partitions: Partitions) -> Result<()>;
-    fn set_paritions_sha(&self, sha: String);
-    fn get_partitions_sha(&self) -> Option<String>;
+    fn add_partitions_sha(&self, sha: String);
```

Comment:
Great.

How will we handle the hash in cluster mode in the future?

Reply:
The partitions SHAs in the context will not be changed when redistributing partitions. We can still get the complete partitions SHAs after building the query pipepline.

Reply:
As we can see the logic tests also passed in `test_sqllogic_base_cluster_linux`.

Reply:
Each node of the cluster has a partially result cache?  Because the partition in the cluster is segment and then read the partitions on the node not the coordinator query node.

Reply:
So here we supposed that the segment hash is not changed the cache results is valid?  The partitions   stored in the meta is segments.

Reply:
> Each node of the cluster has a partially result cache?

No. Result cache is only written by coordinator node after getting all the result blocks.

> So here we supposed that the segment hash is not changed the cache results is valid?

Yes.

It seems that the segment information will also be changed when modifications happen?

Reply:
Yes.
If the partitions changed, the segment will change too, the `segment -> partitions` relation is immutable.